addons/website_hr_recruitment/controllers/main.py

```diff
@@ -25,192 +28,136 @@ def sitemap_jobs(env, rule, qs):
-        if count_remote:
-            count_per_office[None] = count_remote
+        env = request.env(context=dict(request.env.context, show_address=True, no_tag_br=True))
+        website = request.website
+        department = env['hr.department'].browse(to_int(department_id)).exists()
+        country = env['res.country'].browse(to_int(country_id)).exists()
+        office = env['res.partner'].browse(to_int(office_id)).exists()
+        contract_type = env['hr.contract.type'].browse(to_int(contract_type_id)).exists().sudo()
+        industry =  env['res.partner.industry'].browse(to_int(industry_id)).exists().sudo()
 
-        # Employment types
-        if contract_type_id or is_untyped:
-            cross_type_options = options.copy()
-            cross_type_options.update({
-                'allowFuzzy': False,
-                'contract_type_id': None,
-                'is_untyped': False,
-            })
-            cross_type_total, cross_type_details, _ = request.website._search_with_fuzzy("jobs",
-                fuzzy_search_term or search, limit=1000, order="is_published desc, sequence, no_of_recruitment desc",
-                options=cross_type_options)
-            cross_type_jobs = cross_type_details[0].get('results', Jobs)
-        else:
-            cross_type_total = total
-            cross_type_jobs = jobs
-        employment_types = sort(set(j.contract_type_id for j in jobs if j.contract_type_id), 'name')
-        count_per_employment_type = {'all': cross_type_total}
-        for t, jobs_list in groupby(cross_type_jobs, lambda job: job.contract_type_id):
-            count_per_employment_type[t] = len(jobs_list)
-        count_untyped = len(cross_type_jobs.filtered(lambda job: not job.contract_type_id))
-        if count_untyped:
-            count_per_employment_type[None] = count_untyped
+        if not (country or department or office or contract_type or all_countries) \
+            and (code := request.geoip.country_code) \
+                and (country := env['res.country'].search([('code', '=', code)], limit=1)):
+            country_count = env['hr.job'].search_count(AND([
+                website.website_domain(),
+                [('address_id.country_id', '=', country.id)]
+            ]))
+            if not country_count:
+                country = False
 
-        pager = request.website.pager(
+        _total_not_used, details, fuzzy_search_term = website._search_with_fuzzy(
+            "jobs", search,
+            limit=self._jobs_per_page * 50,
+            order="is_published desc, sequence, no_of_recruitment desc",
+            options={
+                'displayDescription': True,
+                'allowFuzzy': not noFuzzy,
+            }
+        )
+        searched_jobs = details[0].get('results', env['hr.job']).sudo()
+        job_filter_values = get_filter_snippets_display_values(searched_jobs)
+        found_jobs = searched_jobs.filtered(job_filtering_condition)
```

Comment:
@Flotchet This looks a bit weird. Normally, the filtering options are given inside the call to `_search_with_fuzzy` which then includes all the necessary domain restrictions inside `_search_get_detail` on the model.
Here, we first determine a fuzzy match on all records - then we filter the records... possibly removing the single one that matched the fuzzy term.